stubs/invoke/invoke/tasks.pyi

```diff
@@ -0,0 +1,72 @@
+from typing import Any
+
+from .config import Config
+from .context import Context
+
+NO_DEFAULT: object
+
+class Task:
+    body: Any
+    __doc__: str
+    __name__: str
+    __module__: Any
+    aliases: Any
+    is_default: Any
+    positional: Any
+    optional: Any
+    iterable: Any
+    incrementable: Any
+    auto_shortflags: Any
+    help: Any
+    pre: Any
+    post: Any
+    times_called: int
+    autoprint: Any
+    def __init__(
+        self,
+        body,
+        name=...,
+        aliases=...,
+        positional=...,
+        optional=...,
+        default: bool = ...,
+        auto_shortflags: bool = ...,
+        help=...,
+        pre=...,
+        post=...,
+        autoprint: bool = ...,
+        iterable=...,
+        incrementable=...,
+    ) -> None: ...
+    @property
+    def name(self): ...
+    def __eq__(self, other): ...
+    def __hash__(self): ...
+    def __call__(self, *args, **kwargs): ...
+    @property
+    def called(self): ...
+    def argspec(self, body): ...
+    def fill_implicit_positionals(self, positional): ...
+    def arg_opts(self, name, default, taken_names): ...
+    def get_arguments(self): ...
+
+# TODO use the args
+def task(*args, **kwargs) -> Task: ...
```

Comment:
If it helps here, I've been using this annotation for `task`:

```python

FuncType = Callable[..., Any]
_F = TypeVar('_F', bound=FuncType)


def task(
	func: _F,
	*,
	name: str = ...,
) -> _F:
	...

```

Reply:
TBH, not sure about this, since it returns a Task object which is not present

Reply:
Haven't looked at this, but this sounds like a use case for ParamSpec from PEP 612.

Reply:
I looked at the implementation in a bit more details (https://github.com/pyinvoke/invoke/blob/master/invoke/tasks.py)

The return object is indeed a `Task`, but the returned type can be controlled via the `klass` kwarg (currently defaults to `Task`). The `Task` object itself implements a `__call__` which ends up calling the wrapped method using the given args/kwargs.

My proposed tweak is accurate for the use-cases I have (naked @task wrapping), but the return type is still not accurate.